website/static/js/registrationUtils.js

```diff
@@ -693,9 +689,20 @@ Draft.prototype.register = function(url, data) {
             }
         })
         .fail(function() {
-            self.onRegisterFail();
+            bootbox.alert({
+                title: 'Registration failed',
+                message: language.registerFail,
+                callback: function() {
+                    $osf.unblock();
+                    if (self.urls.registrations) {
+                        window.location.assign(self.urls.registrations);
+                    }
+                }
+            });
         })
-        .always($osf.unblock);
+        .always(function() {
```

Comment:
not sure why there needs to be another anon function call here, a direct call out to a promise response is 100% ok.